### ISSUE TITLE
Coverity scan, compilers

### ISSUE THREAD

**Issue by LADSoft** [LADSoft/OrangeC#397]
Coverity scan on occ, occil, occpr, and C-style preprocessor

**Comment by GitMensch** [comment]
That milestone is quite far away, maybe it is a good idea to run coverity against the current code base once and inspect for new reported issues?

**Comment by LADSoft** [comment]
opened #589 for this.

**Comment by GitMensch** [comment]
Looks like it was the #581 issue (maybe edit that info in above).
Some of the 96 new items were fixed since the last check, would it be possible to run yesterday's release against Coverity again?

**Comment by LADSoft** [comment]
i'll look at this later...

**Comment by LADSoft** [comment]
this shouldn't have been closed...

**Comment by LADSoft** [comment]
I've submitted this now...   it usually takes a day or two to process on coverity's side...

**Comment by GitMensch** [comment]
The report just got in, so it seems the processing took ~75 minutes only.

Note: as a follow-up to #1003 - 7fb722fedb8a9c387fc5aecd25f7d35bb4a0d427 you possibly want to remove the coff2ieee component. Also note that the dlle + dlpm components were recognized with zero LOC, which should mean that they weren't compiled when the analysis was done - which seems to be related to it not being a common subdirectory to build https://github.com/LADSoft/OrangeC/blob/a345410483c318a309e1d1310c0d9ff008fd4153/src/dirs.mak#L25C8...L25C13

**Comment by LADSoft** [comment]
yeah i haven't deleted any source files for any of that.   In case sometime we decide to revive anything that has been deprecated?   There is a lot of work that went into some of that and there was no point just throwing it out.

At some point when I get time I might make an 'archive' directory and move them there though so it is clear we aren't building them...

**Comment by GitMensch** [comment]
In my opinion: deprecated = still built, obviously needs sources, my not need adjustments (but should still be checked, because their code is used; unused: at least before the next release should be removed from VCS, if it is to be used again one can revert the deletion commit (it is good to have that "clean", possibly for each folder separately, if they don't have a dependency).
David Lindauer ***@***.***> schrieb am Donnerstag, 10. April 2025 um 19:58:

**Comment by LADSoft** [comment]
yeah generally i worry about deleting stuff but yeah I agree it is probably a good idea.    Whenever I go to see what is wrong with the build I will do that...

**Comment by LADSoft** [comment]
Im about halfway through (430 issues left).   I've gone through everything but the compiler.    There are new checks now, it told me to use std::move() a LOT...   I had to make a change to cope with 'const auto &' crashing the compiler...   so...   maybe the libcxx tests will be a little less crashy i don't know...

**Comment by LADSoft** [comment]
over the weekend i got wound up in rewriting the string processing in the name mangling code to prevent buffer overruns.    Which it is always fun to add different ways of doing things into the compiler because sometimes I find bugs :smile:.    

I still managed to get through occparse (leaving only the optimizer/backends to finish up) and there are only 208 issues left...   I should be able to finish that up this week and make another submission...

I haven't looked at appveyor but i got a hint that maybe I've cleaned up the libcxx tests somewhat...   later this week I will reevaluate that...

**Comment by LADSoft** [comment]
i got through all the issues tonite.   And everything still seems to work at this point...   I also fixed occpr and ran the broken code tests.  I suspect changes we made to the preprocessor code at some point broke it due to a bug in the compile scripts for occpr not specifying TARGET_OS_WINDOWS, and also there was a compiler bug in occparse.exe I introduced that caused problems.   I'm doing a final build and then will make another coverity submission if everything is good.

I'm expecting some throw back, one of the issues that I marked as done ended up being intractable   Another I discovered was intractable before I marked it.   I will fix these eventually but for now I'm marking them as not done in a way that I can find them again in the next milestone.

I also did a large amount of refactoring of the compiler(occparse.exe) so we'll see if there is any throw back from that...

**Comment by LADSoft** [comment]
I've taken it as far as I intend to for now....    there are two bugs left open.   I've tried to fix them but am not going to do a submission for it.    I moved some things to pending, meaning I think they may are real issues but trying to fix them now resulted in various compilation problems so I'm opting to leave them broken for now.   I will fix them in the next release.   The rest I left as intentional, but I'll review them at some point to see if I can make code improvements to make them go away...
